upup/pkg/fi/cloudup/dns.go

```diff
@@ -144,6 +144,8 @@ func precreateDNS(cluster *api.Cluster, cloud fi.Cloud) error {
 		return fmt.Errorf("error getting DNS resource records for %q", zone.Name())
 	}
 
+	// TODO: We should change the filter to be a suffix match instead
+	//records, err := rrs.List("", "")
```

Comment:
We have https://github.com/kubernetes/kubernetes/issues/39674 .  It isn't clear that we can implement a suffix match on other providers though...